packages/experiment-tag/src/experiment.ts

```diff
@@ -8,443 +8,675 @@ import {
+   * @param flagKeys
+   */
+  public revertMutations(flagKeys: string[] | undefined = undefined) {
+    if (!flagKeys) {
+      flagKeys = Object.keys(this.appliedMutations);
+    }
+    for (const key of flagKeys) {
+      this.appliedMutations[key]?.forEach((mutationController) => {
+        mutationController.revert();
+      });
+      delete this.appliedMutations[key];
+    }
   }
-  for (const key in variants) {
-    if (flagKeys && !flagKeys.has(key)) {
-      continue;
-    }
-    const variant = variants[key];
-    const isWebExperimentation = variant.metadata?.deliveryMethod === 'web';
-    if (isWebExperimentation) {
-      const shouldTrackExposure =
-        (variant.metadata?.['trackExposure'] as boolean) ?? true;
-      // if payload is falsy or empty array, consider it as control variant
-      const payloadIsArray = Array.isArray(variant.payload);
-      const isControlPayload =
-        !variant.payload || (payloadIsArray && variant.payload.length === 0);
-      if (shouldTrackExposure && isControlPayload) {
-        exposureWithDedupe(key, variant);
+
+  /**
+   * Get redirect URLs for flags.
+   * @param flagKeys
+   */
+  public getRedirectUrls(
+    flagKeys: string[] | undefined = undefined,
+  ): Record<string, Record<string, string>> {
+    const redirectUrlMap: Record<string, Record<string, string>> = {};
+    if (!flagKeys) {
+      flagKeys = Object.keys(this.flagVariantMap);
+    }
+    for (const key of flagKeys) {
+      if (this.flagVariantMap[key] === undefined) {
         continue;
       }
-
-      if (payloadIsArray) {
-        for (const action of variant.payload) {
-          if (action.action === 'redirect') {
-            handleRedirect(action, key, variant);
-          } else if (action.action === 'mutate') {
-            handleMutate(action, key, variant);
-          } else if (action.action === 'inject') {
-            handleInject(action, key, variant);
+      const variants = this.flagVariantMap[key];
+      const redirectUrls = {};
+      Object.keys(variants).forEach((variantKey) => {
+        const variant = variants[variantKey];
+        const payload = variant.payload;
+        if (payload && Array.isArray(payload)) {
+          for (const action of variant.payload) {
+            if (action.action === 'redirect') {
+              const url = action.data?.url;
+              if (url) {
+                redirectUrls[variantKey] = action.data.url;
+              }
+            }
           }
         }
+      });
+      if (Object.keys(redirectUrls).length > 0) {
+        redirectUrlMap[key] = redirectUrls;
       }
     }
+    return redirectUrlMap;
   }
-};
 
-const handleRedirect = (action, key: string, variant: Variant) => {
-  const globalScope = getGlobalScope();
-  if (!globalScope) {
-    return;
+  /**
+   * Preview the effect of a variant on the page.
+   * @param key
+   * @param variant
+   */
+  public previewVariant(key: string, variant: string) {
```

Comment:
Support previewing multiple flags/variants. Maybe we just merge this with applyVariants with another config like `preview`?